MDLIB/AnisotropyRoutines.c

```diff
@@ -349,83 +349,126 @@ double ViscosityConciseAniso( int phase, double lxlz, double lx2, double angle,
 void UpdateAnisoFactor( grid *mesh, mat_prop *materials, params *model, scale *scaling) {
 
   printf("Update anisotropy factor\n");
-  int k, l, c0, Nx, Nz, Ncx, Ncz;
-  Nx  = mesh->Nx;  Ncx = Nx - 1;
-  Nz  = mesh->Nz;  Ncz = Nz-1;
+    int p, k, l, Nx, Nz, Ncx, Ncz, c0, c1;
+  int average = 1;//%model.eta_average; // SHOULD NOT BE ALLOWED TO BE ELSE THAN 1
 
-  // **************** Centroids **************** //
-  #pragma omp parallel for shared( mesh ) private( k, l, c0, ) firstprivate( model, Nx, Ncx, Ncz, materials )
-  for (int k1=0; k1<Ncx*Ncz; k1++ ) {
+  Nx = mesh->Nx;
+  Nz = mesh->Nz;
+  Ncx = Nx-1;
+  Ncz = Nz-1;
 
-    k      = mesh->kp[k1];
-    l      = mesh->lp[k1];
-    c0     = k  + l*(Ncx);
+  // Calculate cell centers shear modulus
+  for ( l=0; l<Ncz; l++ ) {
+    for ( k=0; k<Ncx; k++ ) {
 
-    // Set to 0
-    mesh->aniso_factor_n[c0] = 0.0;
-    
-    // Modify only in active cells
-    if ( mesh->BCp.type[c0] != 30 && mesh->BCp.type[c0] != 31 ) {
+      // Cell center index
+      c0 = k  + l*(Ncx);
 
-      // Loop on phases
-      for (int p=0; p<model->Nb_phases; p++) {
+      // First - initialize to 0
+      if ( model->anisotropy == 1 ) mesh->aniso_factor_n[c0] = 0.0;
```

Comment:
I saw multiple checks for anisotropy, but before calling the function you're checking it anyways. Do you need to add it here?

Reply:
Right, removed superfluous checks